snapshotter/app/service.go

```diff
@@ -156,6 +156,44 @@ func initResolver(config config.Config) (proxyaddress.Resolver, error) {
 	}
 }
 
+type dialerTimeouts struct {
```

Comment:
I don't like the direction it is heading to.

- firecracker-containerd itself is using vsock but we don't have them. What is the difference between firecracker-containerd and demux snapshotter?
- Exposing all knobs to end users basically moving the responsibility from ourselves to them, which isn't good.

Reply:
That's a fair comment. My reasoning for lift and shifting the responsibility was, and based purely on observations, the dial performance seemed to be greatly impacted by host load. I think we could consider mechanisms like exponential backoff to help reduce the number of knobs and loss of performance.